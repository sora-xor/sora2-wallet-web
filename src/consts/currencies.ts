import { Currency } from '@/types/currency';

export const Currencies = [
  { key: Currency.DAI, name: 'Dai', symbol: '$' },
  { key: Currency.USD, name: 'US Dollar', symbol: '$' },
  { key: Currency.AED, name: 'United Arab Emirates Dirham', symbol: 'د.إ' },
  { key: Currency.ARS, name: 'Argentine Peso', symbol: 'ARS$' },
  { key: Currency.AUD, name: 'Australian Dollar', symbol: 'AU$' },
  { key: Currency.BDT, name: 'Bangladeshi Taka', symbol: '৳' },
  { key: Currency.BHD, name: 'Bahraini Dinar', symbol: '.د.ب' },
  { key: Currency.BMD, name: 'Bermudian Dollar', symbol: 'BD$' },
  { key: Currency.GBP, name: 'British Pound Sterling', symbol: '£' },
  { key: Currency.BRL, name: 'Brazil Real', symbol: 'R$' },
  { key: Currency.CAD, name: 'Canadian Dollar', symbol: 'CA$' },
  { key: Currency.CHF, name: 'Switzerland Franc', symbol: 'F' },
  { key: Currency.CLP, name: 'Chilean Peso', symbol: 'CLP$' },
  { key: Currency.CNY, name: 'Chinese yuan', symbol: 'CN¥' },
  { key: Currency.CZK, name: 'Czech Koruna', symbol: 'Kč' },
  { key: Currency.DKK, name: 'Denmark Krone', symbol: 'kr.' },
  { key: Currency.EUR, name: 'Euro', symbol: '€' },
  { key: Currency.GEL, name: 'Georgian Lari', symbol: '₾' },
  { key: Currency.HKD, name: 'Hong Kong Dollar', symbol: 'HK$' },
  { key: Currency.HUF, name: 'Hungarian Forint', symbol: 'Ft' },
  { key: Currency.IDR, name: 'Indonesian Rupiah', symbol: 'Rp' },
  { key: Currency.ILS, name: 'Israel New Shekel', symbol: '₪' },
  { key: Currency.INR, name: 'Indian Rupee', symbol: '₹' },
  { key: Currency.JPY, name: 'Japanese Yen', symbol: '¥' },
  { key: Currency.KRW, name: 'South Korean Won', symbol: '₩' },
  { key: Currency.KWD, name: 'Kuwaiti Dinar', symbol: 'د.ك' },
  { key: Currency.LKR, name: 'Sri Lankan Rupee', symbol: '₨' },
  { key: Currency.MMK, name: 'Myanmar Kyat', symbol: 'K' },
  { key: Currency.MXN, name: 'Mexican Peso', symbol: 'MX$' },
  { key: Currency.MYR, name: 'Malaysian Ringgit', symbol: 'RM' },
  { key: Currency.NGN, name: 'Nigerian Naira', symbol: '₦' },
  { key: Currency.NOK, name: 'Norwegian Krone', symbol: 'kr' },
  { key: Currency.NZD, name: 'New Zealand Dollar', symbol: 'NZ$' },
  { key: Currency.PHP, name: 'Philippine Pesso', symbol: '₱' },
  { key: Currency.PKR, name: 'Pakistani Rupee', symbol: '₨.' },
  { key: Currency.PLN, name: 'Polish Zloty', symbol: 'zł' },
  { key: Currency.RUB, name: 'Russian Ruble', symbol: '₽' },
  { key: Currency.SAR, name: 'Saudi Riyal', symbol: 'ر.س' },
  { key: Currency.SEK, name: 'Swedish Krona', symbol: 'kr' },
  { key: Currency.SGD, name: 'Singapore Dollar', symbol: 'S$' },
  { key: Currency.TRY, name: 'Turkish Lira', symbol: '₺' },
  { key: Currency.THB, name: 'Thai Baht', symbol: '฿' },
  { key: Currency.TWD, name: 'New Taiwan Dollar', symbol: 'NT$' },
  { key: Currency.UAH, name: 'Ukrainian hryvnia', symbol: '₴' },
  { key: Currency.VEB, name: 'Venezuelan bolivar fuerte', symbol: 'Bs.F.' },
  { key: Currency.VND, name: 'Vietnamese dong', symbol: '₫' },
  { key: Currency.XAG, name: 'Silver', symbol: 'XAG' },
  { key: Currency.XAU, name: 'Gold', symbol: 'XAU' },
  { key: Currency.SDR, name: 'IMF Special Drawing Rights', symbol: 'SDR' },
  { key: Currency.ZAR, name: 'South African Rand', symbol: 'R' },
];
