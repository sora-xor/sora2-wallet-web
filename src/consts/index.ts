import type { FPNumber, Operation } from '@sora-substrate/util';

export const HiddenValue = '******';

export const BLOCK_PRODUCE_TIME = 6_000;
export const MAX_ALERTS_NUMBER = 5;

export enum AppWallet {
  GoogleDrive = 'google-drive',
  FearlessWallet = 'fearless-wallet',
  PolkadotJS = 'polkadot-js',
  SubwalletJS = 'subwallet-js',
  TalismanJS = 'talisman',
}

export enum AccountActionTypes {
  Rename = 'rename',
  Export = 'export',
  Logout = 'logout',
  Delete = 'delete',
}

export enum RouteNames {
  InternalConnection = 'InternalConnection',
  WalletConnection = 'WalletConnection',
  WalletSend = 'WalletSend',
  Wallet = 'Wallet',
  WalletAssetDetails = 'WalletAssetDetails',
  CreateToken = 'CreateToken',
  ReceiveToken = 'ReceiveToken',
  AddAsset = 'AddAsset',
  SelectAsset = 'SelectAsset',
}

export enum WalletTabs {
  Assets = 'WalletAssets',
  History = 'WalletHistory',
}

export enum TokenTabs {
  Token = 'CreateSimpleToken',
  NonFungibleToken = 'CreateNftToken',
}

export enum AddAssetTabs {
  Token = 'AddAssetToken',
  NFT = 'AddAssetNFT',
}

export enum WalletFilteringOptions {
  All = 'All',
  Currencies = 'Currencies',
  NFT = 'NFT',
}

export enum SoraNetwork {
  Dev = 'Dev',
  Test = 'Test',
  Stage = 'Stage',
  Prod = 'Prod',
}

export enum HashType {
  ID = 'id',
  Block = 'block',
  Account = 'account',
  EthAccount = 'ethAccount',
  EthTransaction = 'ethTransaction',
}

export enum ExplorerType {
  Sorascan = 'sorascan',
  Subscan = 'subscan',
  Polkadot = 'polkadot',
}

export type ExplorerLink = {
  type: ExplorerType;
  value: string;
};

export enum Step {
  CreateSimpleToken = 'CreateSimpleToken',
  ConfirmSimpleToken = 'ConfirmSimpleToken',
  CreateNftToken = 'CreateNftToken',
  ConfirmNftToken = 'ConfirmNftToken',
  Warn = 'Warn',
}

export enum LoginStep {
  Welcome = 'Welcome',
  Import = 'Import',
  ImportCredentials = 'Import/Credentials',
  ImportExternalExtensionList = 'ImportExternal/ExtensionList',
  ImportExternalAcccountList = 'ImportExternal/AccountList',
  SeedPhrase = 'Create/SeedPhrase',
  ConfirmSeedPhrase = 'Create/ConfirmSeedPhrase',
  CreateCredentials = 'Create/Credentials',
  AccountList = 'AccountList',
}

export const AccountImportInternalFlow = [LoginStep.Import, LoginStep.ImportCredentials];
export const AccountImportExternalFlow = [LoginStep.ImportExternalExtensionList, LoginStep.ImportExternalAcccountList];
export const AccountCreateFlow = [LoginStep.SeedPhrase, LoginStep.ConfirmSeedPhrase, LoginStep.CreateCredentials];

export interface WalletPermissions {
  addAssets?: boolean;
  addLiquidity?: boolean;
  bridgeAssets?: boolean;
  createAssets?: boolean;
  sendAssets?: boolean;
  showAssetDetails?: boolean;
  swapAssets?: boolean;
}

export type EthBridgeUpdateHistory = (updateWalletHistory: VoidFunction) => Promise<void>;

export interface WalletAssetFilters {
  option: WalletFilteringOptions;
  verifiedOnly: boolean;
  zeroBalance: boolean;
}

export interface WalletInitOptions {
  withoutStore?: boolean;
  permissions?: WalletPermissions;
  updateEthBridgeHistory?: EthBridgeUpdateHistory;
}

export interface NetworkFeeWarningOptions {
  type: Operation;
  isXor?: boolean;
  amount?: FPNumber;
}

export enum FontSizeRate {
  SMALL = 'small',
  MEDIUM = 'medium',
  NORMAL = 'normal',
}

export enum FontWeightRate {
  SMALL = 'small',
  MEDIUM = 'medium',
  NORMAL = 'normal',
}

export enum LogoSize {
  MINI = 'mini',
  SMALL = 'small',
  MEDIUM = 'medium',
  BIG = 'big',
  BIGGER = 'bigger',
  LARGE = 'large',
}

export enum PaginationButton {
  First = 'first',
  Prev = 'prev',
  Next = 'next',
  Last = 'last',
}

// states for Ethereum Bridge transaction
export enum ETH_BRIDGE_STATES {
  INITIAL = 'INITIAL',
  SORA_SUBMITTED = 'SORA_SUBMITTED',
  SORA_PENDING = 'SORA_PENDING',
  SORA_REJECTED = 'SORA_REJECTED',
  SORA_COMMITED = 'SORA_COMMITED',
  EVM_SUBMITTED = 'EVM_SUBMITTED',
  EVM_PENDING = 'EVM_PENDING',
  EVM_REJECTED = 'EVM_REJECTED',
  EVM_COMMITED = 'EVM_COMMITED',
}

export const ObjectInit = () => null;

/**
 * DO NOT IMPORT THIS CONST!
 *
 * Contains wallet-specific words which shouldn't be translated.
 * It's used in TranslationMixin of SORA Wallet project and it's extended in Polkaswap TranslationMixin.
 */
export const TranslationConsts = {
  Ethereum: 'Ethereum',
  Etherscan: 'Etherscan',
  Hashi: 'HASHI',
  PolkadotJs: 'Polkadot{.js}',
  Sora: 'SORA',
  TBC: 'TBC',
  XYK: 'XYK',
  NFT: 'NFT',
  Polkadot: 'Polkadot',
  SORAScan: 'SORAScan',
  Subscan: 'Subscan',
  QR: 'QR', // QR code
  IPFS: 'IPFS', // Inter-planetary file system
  // SORA networks
  soraNetwork: {
    Dev: 'SORA Devnet',
    Test: 'SORA Testnet (private)',
    Stage: 'SORA Testnet',
    Prod: 'SORA Mainnet',
  },
  JSON: '.json',
} as const;
