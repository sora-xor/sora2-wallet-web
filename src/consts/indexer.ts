export const wsClientRetryAttempts = 10;
export const wsClientLazy = true;
export const wsClientReconnect = true;
