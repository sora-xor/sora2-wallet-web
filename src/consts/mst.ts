export const mstTrxDeadline = {
  '1D': 14400, // 1 Day
  '7D': 100800, // 7 Days
  '14D': 201600, // 14 Days
  '30D': 432000, // 30 Days
};
