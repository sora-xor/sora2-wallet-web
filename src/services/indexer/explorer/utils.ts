import { formatStringNumber } from '../../../util';

import type { SubqueryAssetEntity } from '../subquery/types';
import type { SubsquidAssetEntity } from '../subsquid/types';
import type { FiatPriceObject, UpdatesStream, PoolApyObject } from '../types';

export function parseAssetFiatPrice(entity: SubsquidAssetEntity | SubqueryAssetEntity): FiatPriceObject {
  const acc = {};
  const id = entity.id;
  const priceFPNumber = formatStringNumber(entity.priceUSD);
  const isPriceFinity = priceFPNumber.isFinity();
  if (isPriceFinity) {
    acc[id] = priceFPNumber.toCodecString();
  }
  return acc;
}

export function parsePriceStreamUpdate(entity: UpdatesStream): Nullable<FiatPriceObject> {
  if (!entity?.data) return null;

  const data = JSON.parse(entity.data);

  return Object.entries(data).reduce((acc, [id, price]) => {
    const priceFPNumber = formatStringNumber(price as string);
    const isPriceFinity = priceFPNumber.isFinity();
    if (isPriceFinity) {
      acc[id] = priceFPNumber.toCodecString();
    }
    return acc;
  }, {});
}

export function parseApyStreamUpdate(entity: UpdatesStream): PoolApyObject {
  const data = entity?.data ? JSON.parse(entity.data) : {};

  return Object.entries(data).reduce((acc, [id, apy]) => {
    const strategicBonusApyFPNumber = formatStringNumber(apy as string);
    const isStrategicBonusApyFinity = strategicBonusApyFPNumber.isFinity();
    if (isStrategicBonusApyFinity) {
      acc[id] = strategicBonusApyFPNumber.toCodecString();
    }
    return acc;
  }, {});
}
