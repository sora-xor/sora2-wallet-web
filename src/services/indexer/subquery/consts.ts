export const SubstrateEvents = {
  CurrenciesTransferred: {
    method: 'Transferred',
    section: 'currencies',
  },
};
