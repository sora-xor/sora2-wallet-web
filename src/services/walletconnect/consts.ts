export const WALLET_CONNECT_PROJECT_ID = 'feeab08b50e0d407f4eb875d69e162e8';
