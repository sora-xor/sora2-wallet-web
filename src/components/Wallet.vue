<template>
  <wallet-base :title="headerTitle" :show-back="!!selectedTransaction" :reset-focus="headerTitle" @back="handleBack">
    <template v-if="!selectedTransaction" #actions>
      <s-button :type="isMultisig() ? 'primary' : 'tertiary'" @click="handleMST"> Multi-Sig </s-button>

      <s-button type="action" :tooltip="t('accountSettings.title')" @click="handleAccountSettings">
        <s-icon name="basic-settings-24" size="28" />
      </s-button>
      <s-button
        v-if="permissions.createAssets"
        type="action"
        :tooltip="t('createTokenText')"
        @click="handleCreateToken"
      >
        <s-icon name="various-atom-24" size="28" />
      </s-button>
    </template>

    <wallet-account v-if="!selectedTransaction">
      <qr-code-scan-button alternative @change="parseQrCodeValue" />

      <s-button
        type="action"
        size="small"
        alternative
        rounded
        :tooltip="t('code.receive')"
        @click="receiveByQrCode(null)"
      >
        <s-icon name="finance-receive-show-QR-24" size="24" />
      </s-button>

      <s-button
        type="action"
        size="small"
        alternative
        rounded
        :tooltip="t('account.switch')"
        @click="handleSwitchAccount"
      >
        <s-icon name="arrows-refresh-ccw-24" size="24" />
      </s-button>

      <account-actions-menu v-if="!isExternal" :actions="accountActions" @select="handleAccountActionType" />
    </wallet-account>

    <div v-show="!selectedTransaction" class="wallet">
      <s-tabs v-model="currentTab" type="rounded">
        <s-tab v-for="tab in WalletTabs" :key="tab" :label="t(`wallet.${tab}`)" :name="tab" />
      </s-tabs>
      <component :is="currentTab" @swap="handleSwap" />
    </div>

    <wallet-transaction-details v-if="selectedTransaction" />

    <account-settings-dialog :visible.sync="accountSettingsVisibility" />
    <mst-onboarding-dialog :visible.sync="mstOnboardingDialog" />
    <multisig-change-name-dialog :visible.sync="dialogMSTNameChange" />

    <template v-if="!isExternal">
      <account-rename-dialog
        :visible.sync="accountRenameVisibility"
        :loading="loading"
        @confirm="handleAccountRename"
      />
      <account-export-dialog
        :visible.sync="accountExportVisibility"
        :loading="loading"
        @confirm="handleAccountExport"
      />
      <account-delete-dialog
        :visible.sync="accountDeleteVisibility"
        :loading="loading"
        @confirm="handleAccountDelete"
      />
    </template>
  </wallet-base>
</template>

<script lang="ts">
import { api } from '@sora-substrate/sdk';
import { Component, Mixins } from 'vue-property-decorator';

import { RouteNames, WalletTabs, AccountActionTypes } from '../consts';
import { state, getter, mutation } from '../store/decorators';

import AccountActionsMenu from './Account/ActionsMenu.vue';
import AccountExportDialog from './Account/ConfirmDialog.vue';
import AccountDeleteDialog from './Account/DeleteDialog.vue';
import AccountRenameDialog from './Account/RenameDialog.vue';
import AccountSettingsDialog from './Account/SettingsDialog.vue';
import WalletAccount from './Account/WalletAccount.vue';
import AccountActionsMixin from './mixins/AccountActionsMixin';
import OperationsMixin from './mixins/OperationsMixin';
import QrCodeParserMixin from './mixins/QrCodeParserMixin';
import MstOnboardingDialog from './MST/MstOnboardingDialog.vue';
import MultisigChangeNameDialog from './MST/MultisigChangeNameDialog.vue';
import QrCodeScanButton from './QrCode/QrCodeScanButton.vue';
import WalletAssets from './WalletAssets.vue';
import WalletBase from './WalletBase.vue';
import WalletHistory from './WalletHistory.vue';
import WalletTransactionDetails from './WalletTransactionDetails.vue';

import type { WalletPermissions } from '../consts';
import type { HistoryItem } from '@sora-substrate/sdk';

@Component({
  components: {
    WalletBase,
    WalletAccount,
    WalletAssets,
    WalletHistory,
    QrCodeScanButton,
    WalletTransactionDetails,
    AccountActionsMenu,
    AccountRenameDialog,
    AccountExportDialog,
    AccountDeleteDialog,
    AccountSettingsDialog,
    MstOnboardingDialog,
    MultisigChangeNameDialog,
  },
})
export default class Wallet extends Mixins(AccountActionsMixin, OperationsMixin, QrCodeParserMixin) {
  readonly WalletTabs = WalletTabs;
  readonly accountActions = [
    AccountActionTypes.Rename,
    AccountActionTypes.Export,
    AccountActionTypes.Logout,
    AccountActionTypes.Delete,
  ];

  @state.router.currentRouteParams private currentRouteParams!: Record<string, Nullable<WalletTabs>>;
  @state.settings.permissions permissions!: WalletPermissions;
  @state.settings.isMSTAvailable isMSTAvailable!: boolean;
  @state.account.isExternal isExternal!: boolean;
  @state.account.isMST isMST!: boolean;
  @state.account.isMstAddressExist isMstAddressExist!: boolean;

  @getter.transactions.selectedTx selectedTransaction!: Nullable<HistoryItem>;

  @mutation.transactions.resetTxDetailsId private resetTxDetailsId!: FnWithoutArgs;

  currentTab: WalletTabs = WalletTabs.Assets;

  accountSettingsVisibility = false;
  mstOnboardingDialog = false;
  dialogMSTNameChange = false;

  get headerTitle(): string {
    if (!this.selectedTransaction) return this.t('account.accountTitle');

    return this.getTitle(this.selectedTransaction);
  }

  get isMSTAccount(): boolean {
    return this.isMST && api.mst.getMSTName() !== '';
  }

  get hasMSTAccount(): boolean {
    return !this.isMST && (this.isMstAddressExist || api.mst.getMSTName() !== '');
  }

  mounted(): void {
    if (this.currentRouteParams.currentTab) {
      this.currentTab = this.currentRouteParams.currentTab;
    }
  }

  handleSwap(asset: any): void {
    this.$emit('swap', asset);
  }

  handleCreateToken(): void {
    this.navigate({ name: RouteNames.CreateToken });
  }

  handleSwitchAccount(): void {
    this.navigate({ name: RouteNames.WalletConnection });
  }

  handleMST(): void {
    if (this.isMSTAccount) {
      // User is currently in MST account
      this.dialogMSTNameChange = true;
    } else if (this.hasMSTAccount) {
      // User has an MST account but is not currently in it
      this.dialogMSTNameChange = true;
    } else {
      // User does not have an MST account
      this.mstOnboardingDialog = true;
    }
  }

  handleAccountActionType(actionType: string) {
    this.handleAccountAction(actionType, this.account);
  }

  handleAccountSettings(): void {
    this.accountSettingsVisibility = !this.accountSettingsVisibility;
  }

  handleBack(): void {
    if (this.selectedTransaction) {
      this.resetTxDetailsId();
    }
  }

  isMultisig(): boolean {
    return this.isMSTAccount;
  }
}
</script>

<style lang="scss">
.wallet {
  @include custom-tabs;
}
</style>

<style scoped lang="scss">
.wallet {
  margin-top: #{$basic-spacing-medium};
}
</style>
