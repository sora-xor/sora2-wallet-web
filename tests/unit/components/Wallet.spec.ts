import Vuex from 'vuex'
import { shallowMount } from '@vue/test-utils'

import { useDescribe, localVue } from '../../utils'
import { MOCK_HISTORY } from '../../utils/mock'

import Wallet from '@/components/Wallet.vue'
import { WalletTabs } from '@/consts'

const createStore = (currentTab: WalletTabs) => new Vuex.Store({
  modules: {
    Account: {
      getters: {
        account: () => ({
          address: 'dfsakljfdlkjfhfkjladshslfjafds',
          name: 'Mock',
          password: '123qwaszx'
        }),
        activity: () => MOCK_HISTORY
      },
      actions: {
        getAccountActivity: jest.fn()
      }
    },
    Router: {
      getters: {
        currentRouteParams: () => ({
          currentTab
        })
      },
      actions: {
        navigate: jest.fn()
      }
    }
  } as any
})

useDescribe('Wallet.vue', Wallet, () => {
  Object.values(WalletTabs).map(item => it(`[WalletTabs.${item}]: should be rendered correctly`, () => {
    const wrapper = shallowMount(Wallet, { localVue, store: createStore(item) })
    expect(wrapper.element).toMatchSnapshot()
  }))
})
