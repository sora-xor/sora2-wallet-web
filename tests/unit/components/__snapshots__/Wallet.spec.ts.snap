// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet.vue [WalletTabs.WalletActivity]: should be rendered correctly 1`] = `
<wallet-base-stub
  actionicon="various-atom-24"
  actiontooltip="wallet.createToken"
  showaction="true"
  showheader="true"
  title=""
  tooltip=""
>
  <wallet-account-stub
    showcontrols="true"
  />
   
  <div
    class="wallet"
  >
    <s-tabs-stub
      borderradius="small"
      position="top"
      type="rounded"
      usedesignsystem="true"
      value="WalletAssets"
    >
      <s-tab-stub
        name="WalletAssets"
      />
      <s-tab-stub
        name="WalletActivity"
      />
    </s-tabs-stub>
     
    <walletassets-stub />
  </div>
</wallet-base-stub>
`;

exports[`Wallet.vue [WalletTabs.WalletAssets]: should be rendered correctly 1`] = `
<wallet-base-stub
  actionicon="various-atom-24"
  actiontooltip="wallet.createToken"
  showaction="true"
  showheader="true"
  title=""
  tooltip=""
>
  <wallet-account-stub
    showcontrols="true"
  />
   
  <div
    class="wallet"
  >
    <s-tabs-stub
      borderradius="small"
      position="top"
      type="rounded"
      usedesignsystem="true"
      value="WalletAssets"
    >
      <s-tab-stub
        name="WalletAssets"
      />
      <s-tab-stub
        name="WalletActivity"
      />
    </s-tabs-stub>
     
    <walletassets-stub />
  </div>
</wallet-base-stub>
`;
